packages/jira-adapter/src/client/connection.ts

```diff
@@ -17,11 +17,26 @@ import { AccountId } from '@salto-io/adapter-api'
 import { client as clientUtils } from '@salto-io/adapter-components'
 import { Credentials } from '../auth'
 
+const isValidCredentials = async (connection: clientUtils.APIConnection): Promise<boolean> =>
+  connection.get('/rest/api/3/configuration')
+    .then(
+      () => true,
+      () => false
+    )
+
+const getBaseUrl = async (connection: clientUtils.APIConnection): Promise<string> => {
+  const response = await connection.get('/rest/api/3/serverInfo')
+  return response.data.baseUrl
+}
+
 export const validateCredentials = async (
   { connection }: { connection: clientUtils.APIConnection },
 ): Promise<AccountId> => {
-  const res = await connection.get('/rest/api/3/serverInfo')
-  return res.data.baseUrl
+  const isValid = await isValidCredentials(connection)
+  if (!isValid) {
+    return Promise.reject(new Error('Invalid Credentials'))
```

Comment:
This is the same as simply throwing an error
```suggestion
   throw new Error('Invalid Credentials')
```

Reply:
I've got warning when running my playground that stated that in future release of ES-lint this will be an error. After reading online, using `Promise.reject()` is safer than throwing an error directly.
You can refer to [this](https://stackoverflow.com/questions/33445415/javascript-promises-reject-vs-throw#:~:text=Any%20time%20you%20are%20inside,callback%2C%20you%20must%20use%20reject%20.) discussion